### PR TITLE
fix(gatsby): parse pageData response when contentType is not set

### PR DESCRIPTION
**Pull request by wardpeet**
## Description
Not all hosting providers set a content-type of `application/json`. We only accept page-data that have that content-type. This PR will also parse responeText and look if it has a proper pageData object. If it does, it's considered json.

## Related Issues
https://github.com/gatsbyjs/gatsby/issues/15151

### CODE REVIEW

**Comment by tahasabih** [comment]
Would be great if this can be merged and released! :) 
Had to downgrade our gatsby version on prod because it stopped working on s3 :( This seems to be a good fix!

File: packages/gatsby/cache-dir/loader.js [hunk 1]
```diff
@@ -34,16 +34,23 @@ const loadPageDataJson = loadObj => {
   const url = createPageDataUrl(pagePath)
   return doFetch(url).then(req => {
     const { status, responseText } = req
+    let isJson = false
```
Comment (pieh) [inline packages/gatsby/cache-dir/loader.js]: NIT: this is never set to anything else and is used later in condition in https://github.com/gatsbyjs/gatsby/blob/482e4829f670c6bb910e4610b4db4e23b14dde1d/packages/gatsby/cache-dir/loader.js#L57 (there are no other usage of `isJson`) - we could remove this completely, right?

Comment (wardpeet) [reply packages/gatsby/cache-dir/loader.js]: that's true! thanks!

**Comment by pieh** [comment]
Published `gatsby@2.13.6`

**Pull request merged**